File: effective/src/com/lab/java/zayson/item52/NexonGames.java
package com.lab.java.zayson.item52;

public class NexonGames {
    String title() {
        return "넥슨 게임";
    }
}

class MapleStory extends NexonGames {
    @Override
    String title() {
        return "메이플스토리";
    }
}

class FifaOnline4 extends NexonGames {
    @Override
    String title() {
        return "피파온라인 4";
    }
}

File: practice/src/main/java/zayson/java/lab/kotlin/grammer/lecture14/NexonGames.kt
package zayson.java.lab.kotlin.grammer.lecture14

sealed class NexonGames (
    val name: String,
    val year: Int
)

// 상속
class MapleStory : NexonGames("메이플 스토리", 2003)

class FifaOnline4 : NexonGames("피파 온라인 4", 2018)
